frontend/src/lib/components/PayGateMini/PayGateMiniButton.tsx

```diff
@@ -5,22 +5,28 @@ import { billingProductLogic } from 'scenes/billing/billingProductLogic'
 import { PlanComparisonModal } from 'scenes/billing/PlanComparison'
 import { urls } from 'scenes/urls'
 
-import { BillingProductV2Type, BillingV2FeatureType } from '~/types'
+import { BillingProductV2AddonType, BillingProductV2Type, BillingV2FeatureType } from '~/types'
 
 export const PayGateMiniButton = ({
     product,
     gateVariant,
     featureInfo,
     onClick,
 }: {
-    product: BillingProductV2Type
+    product: BillingProductV2Type | BillingProductV2AddonType
     featureInfo: BillingV2FeatureType
     gateVariant: 'add-card' | 'contact-sales' | 'move-to-cloud'
     onClick?: () => void
 }): JSX.Element => {
     const { isPlanComparisonModalOpen } = useValues(billingProductLogic({ product }))
     const { toggleIsPlanComparisonModalOpen } = useActions(billingProductLogic({ product }))
 
+    // We know that the product is a BillingProductV2Type because that's the only
+    // type that can be used here from the PayGateMini component. But TypeScript doesn't
+    // know that, so we need to cast it to the correct type, and it's simpler to do this
+    // here than in the PayGateMini component.
+    const typedProduct = product as BillingProductV2Type
```

Comment:
I'm confused why did you change the product type at the top then?

Reply:
Because if I didn't then I'd get a type error in the PayGateMini component. But in that component I _can_ have either an addon or a regular product type. TBH idk why the addon type is different from the regular product type...